src/hotspot/os/linux/attachListener_linux.cpp

```diff
@@ -266,21 +266,21 @@ LinuxAttachOperation* LinuxAttachListener::read_request(int s) {
   // Read until all (expected) strings have been read, the buffer is
   // full, or EOF.
 
-  int off = 0;
-  int left = max_len;
+  size_t off = 0;
+  size_t left = max_len;
```

Comment:
It might be better to use ssize_t here, because of code like
` left -= n;`
below.  If that underflows as unsigned then the loop won't terminate like it should.

Reply:
Theoretically but we assert that n <= left.